modules/mod-deep-learning/CMakeLists.txt

```diff
@@ -0,0 +1,29 @@
+set( SOURCES
```

Comment:
I would like every new CMakeLists.txt file for a library or a module to begin with long-form comments at the top giving some overview of the purpose:
```
#[[
...
]]
```

You might explain what some of the individual source files do but it is not necessary for all of them.

Reply:
Before diving deeply in these new source code files to understand them, the first thing I did was look through the #include directives to understand their dependencies, so that I could understand how to approach understanding bottom-up or top-down.  I made this graph with Graphviz of dependencies among the .cpp/.h file pairs:

![graph dot](https://user-images.githubusercontent.com/11670369/148216264-99ca6b94-2470-418b-abfc-6cd4d1afefca.gif)

Ideally dependencies among source code files should have no cycles, but I find two non-trivial strongly connected components here, as boxes.

Can these dependencies be broken?  Maybe my further exploration will tell me.